README.md

```diff
@@ -524,3 +524,88 @@ Returns the number of bins according to [Scott’s normal reference rule](https:
+
+If *valueX* is specified, sets the value accessor for the X value to the specified function or constant and returns this bin2d generator. If *valueX* is not specified, returns the current value accessor, which defaults to the identity function.
+
+When a bin2d is [generated](#_bin2d), the value accessor will be invoked for each element in the input data array, being passed the element `d`, the index `i`, and the array `data` as three arguments. The default value accessor assumes that the input data are orderable (comparable), such as numbers or dates. If your data are not, then you should specify an accessor that returns the corresponding orderable value for a given datum.
+
+This is similar to mapping your data to values before invoking the bin2d generator, but has the benefit that the input data remains associated with the returned bins, thereby making it easier to access other fields of the data.
+
+<a name="bin2d_domain_x" href="#bin2d_domain_x">#</a> <i>bin2d</i>.<b>domainX</b>([<i>domain</i>]) [<>](https://github.com/d3/d3-array/blob/master/src/bin2d.js#L62 "Source")
+
+If *domainX* is specified, sets the X domain accessor to the specified function or array and returns this bin2d generator. If *domainX* is not specified, returns the current domain accessor, which defaults to [extentX](#extent_x). The bin2d domain is defined as an array [*min*, *max*], where *min* is the minimum observable X value and *max* is the maximum observable Y value; both values are inclusive. Any value outside of this domain will be ignored when the bin2d is [generated](#_bin2d).
+
+For example, if you are using the the bin2d in conjunction with a [linear scale](https://github.com/d3/d3-scale/blob/master/README.md#linear-scales) `x`, you might say:
+
+```js
+var bin2d = d3.bin2d()
+    .domainX(x.domain())
+    .thresholdsX(x.ticks(20));
+```
+
+<a name="bin2d_domain_x" href="#bin2d_domain_x">#</a> <i>bin2d</i>.<b>domainY</b>([<i>domain</i>]) [<>](https://github.com/d3/d3-array/blob/master/src/bin2d.js#L62 "Source")
+
+If *domainY* is specified, sets the X domain accessor to the specified function or array and returns this bin2d generator. If *domainY* is not specified, returns the current domain accessor, which defaults to [extentX](#extent_x). The bin2d domain is defined as an array [*min*, *max*], where *min* is the minimum observable X value and *max* is the maximum observable Y value; both values are inclusive. Any value outside of this domain will be ignored when the bin2d is [generated](#_bin2d).
+
+For example, if you are using the the bin2d in conjunction with a [linear scale](https://github.com/d3/d3-scale/blob/master/README.md#linear-scales) `x`, you might say:
+
+```js
+var bin2d = d3.bin2d()
+    .domainY(x.domain())
+    .thresholdsX(x.ticks(20));
+```
+
+You can then compute the bins from an array of numbers like so:
```

Comment:
Realistically, you would not use `bin2d` if you have an array of numbers. I suggest to rework this to use an array of objects, from which X and Y can both be derived.

Reply:
I am not sure why one would prevail over the other. And having an object would require to define default properties e.g. x,y for {x:0,y:0} or col1,col2 for {col1:0,col2:0} or any other convention. We are more likely to have an agnostic array [0,0] that would come naturally from a tabular source.